### DISCUSSION TITLE
issue: Dynamically field won't work as expected

### DISCUSSION THREAD

**Discussion by phongtruonghung** [react-hook-form/react-hook-form#6932] [Q&A]
### Version Number

7.18.0

### Codesandbox/Expo snack

https://codesandbox.io/s/nervous-frost-5glqz?file=/src/App.js

### Steps to reproduce

1. Create a form which have 2 fields `foo` and `bar`, each field has its own value
2. Create a simple input and register it to a variable `field` which created using `useState` hook, so it can be changed between `foo` and `bar`. Default value of `field` is `foo`
3. Switch back and forth `field` between `foo` and `bar` using a simple button

Now just click button several times, and you can see value of input doesn't changed corresponding to `field`, it stops on value of `bar` permanently.

### Expected behaviour

Input value must changed corresponding to `field`
When `field` is `foo`, input must show exact value of `foo`
The same with `bar`

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
couple suggestions here: 

1. if you have two fields, better to register them separately.
2. use `setValue` instead of `setState`

**Comment by slbls** [comment]
I've run into this before and one workaround is to add a `key` prop that reflects the name of the field. This of course comes with the downside of rerendering the component every time the name changes, but it does maintain the proper value associated with each name.

**Reply by bluebill1049** [reply]
thanks. this is been documented in the FAQ as well: https://react-hook-form.com/faqs#Whyisdefaultvaluenotchangingcorrectlywithternaryoperator
